src/common/core/corerel.js

```diff
@@ -171,17 +137,73 @@ define([
                 return false;
             }
         };
+
+        this.persist = function (node) {
+            var persisted = innerCore.persist(node);
+
+            storeNewInverseOverlays(self.getRoot(node));
+
+            return persisted;
+        };
         //</editor-fold>
 
         //<editor-fold=Added Methods>
+        this.getInverseOverlayOfNode = function (node) {
+            var hash = self.getHash(node),
```

Comment:
It should be more of a look-up.
The normalization in most of the cases should not be computationally heavy, but will remove after the first branch just in case.